File: src/main/java/com/prmncr/normativecontrol/components/CorrectDocumentParams.java
package com.prmncr.normativecontrol.components;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import java.math.BigInteger;

@Component
public class CorrectDocumentParams {
    @Value("${document.page.width}")
    public BigInteger pageWidth;
    @Value("${document.page.height}")
    public BigInteger pageHeight;

    @Value("${document.page.margin.top}")
    public BigInteger pageMarginTop;
    @Value("${document.page.margin.right}")
    public BigInteger pageMarginRight;
    @Value("${document.page.margin.bottom}")
    public BigInteger pageMarginBottom;
    @Value("${document.page.margin.left}")
    public BigInteger pageMarginLeft;
}

File: src/main/kotlin/com/prmncr/normativecontrol/components/CorrectDocumentParams.kt
package com.prmncr.normativecontrol.components

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class CorrectDocumentParams {
    @Value("\${document.page.width}")
    var pageWidth: Long = 0

    @Value("\${document.page.height}")
    var pageHeight: Long = 0

    @Value("\${document.page.margin.top}")
    var pageMarginTop = 0.0

    @Value("\${document.page.margin.right}")
    var pageMarginRight = 0.0

    @Value("\${document.page.margin.bottom}")
    var pageMarginBottom = 0.0

    @Value("\${document.page.margin.left}")
    var pageMarginLeft = 0.0
}